ms-internal/git-merge.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+BRANCH=${1:-"auto-pr-$RANDOM"}
+BASE=${2:-"dev"}
+UPSTREAM=${3:-"Green-Software-Foundation/carbon-aware-sdk"}
+
+UPSTREAMREPO="https://github.com/$UPSTREAM"
+COMMITMESSAGE="Warning: git push of upstream contents failed, saving error info into an empty-PR for investigation"
+STATUS=0
+
+git config user.name "GitHub Actions Bot"
+git config user.email "<>"
+
+# upstream contents needed for creating the PR
+git remote add upstream $UPSTREAMREPO
+git fetch upstream
+git checkout -b $BRANCH upstream/$BASE 
+
+# git push, capturing output text and exit code
+echo "Creating a PR with fetched-contents from upstream:$BASE into origin:$BASE."
+GIT_PUSH_OUTPUT=$(git push --set-upstream origin $BRANCH 2>&1) # without 'workflow' permission this fails if changes to .github\workflow\*.yml file(s) are present
```

Comment:
Suggestion: Should a 'dry run' performed first to see what changes are going to go, so it would be easier to have a list of files that might be in conflict? This is just for troubleshooting purposes. You could have this as an artifact.

Reply:
hi @juzuluag , there are actually no conflict at this point, ever, since we're only fetching upstream, and create a PR from it. No `git merge` attempt occurs before the PR is created.